### DISCUSSION TITLE
Fallback image loading

### DISCUSSION THREAD

**Discussion by SelfMadeSystem** [vercel/next.js#70113] [Help]
### Summary

What I want: to fetch an image from an image provider, return it if it exists, else return a fallback image.

What I tried:

1.

This worked very well, but it doesn't allow for a fallback in case of a 404 (afaik)
```ts
    async rewrites() {
        return [
            {
                source: '/api/img/:type/:id/:nou*',
                destination: `${IMG_API}/:type/:id`,
            },
        ];
    },
```

2.

`api/[...a]`

I tried doing something like this

```ts
import fallback from '@assets/fallback.jpg';
import type { NextRequest } from 'next/server';

const IMG_API = process.env.IMG_ENDPOINT;

// source: '/api/img/:type/:id/:nou*',
// destination: `${IMG_API}/:type/:id`,
export async function GET(req: NextRequest) {
    const [type, id] = req.nextUrl.pathname.split('/').slice(3);

    const fetchRes = await fetch(`${IMG_API}/${type}/${id}`);

    if (fetchRes.ok) {
        return fetchRes;
    }

    return new Response(fallback, { headers: { 'content-type': 'image/png' } });
}
```

However, I cannot pass the `fallback` import to the `Response` object.

Btw, I put `fallback.jpg` in the `assets` folder.

I found https://github.com/vercel/next.js/discussions/15453, but this doesn't seem to apply to my case as `fallback` is in the `assets` folder and I might not know where it exists during runtime.

**Comment by Juneezee** [comment]
I think this StackOverflow answer is exactly what you want https://stackoverflow.com/a/66953317
